src/app/clusters/actions-server/actions-server.h

```diff
@@ -0,0 +1,370 @@
+    void HandleEnableAction(HandlerContext & ctx, const Commands::EnableAction::DecodableType & commandData);
+    void HandleEnableActionWithDuration(HandlerContext & ctx,
+                                        const Commands::EnableActionWithDuration::DecodableType & commandData);
+    void HandleDisableAction(HandlerContext & ctx, const Commands::DisableAction::DecodableType & commandData);
+    void HandleDisableActionWithDuration(HandlerContext & ctx,
+                                         const Commands::DisableActionWithDuration::DecodableType & commandData);
+};
+
+class Delegate
+{
+public:
+    virtual ~Delegate() = default;
+
+    /**
+     * Get the action at the Nth index from list of actions.
+     * @param index The index of the action to be returned. It is assumed that actions are indexable from 0 and with no gaps.
+     * @param action A reference to the ActionStructStorage which should be initialized via copy/assignments or calling Set().
+     * @return Returns a CHIP_NO_ERROR if there was no error and the action was returned successfully.
+     * CHIP_ERROR_PROVIDER_LIST_EXHAUSTED if the index is past the end of the list of actionList.
+     */
+    virtual CHIP_ERROR ReadActionAtIndex(uint16_t index, ActionStructStorage & action) = 0;
+
+    /**
+     * Get the EndpointList at the Nth index from list of endpointList.
+     * @param index The index of the endpointList to be returned. It is assumed that actions are indexable from 0 and with no gaps.
+     * @param action A reference to the EndpointListStorage which should be initialized via copy/assignments or calling Set().
+     * @return Returns a CHIP_NO_ERROR if there was no error and the epList was returned successfully.
+     * CHIP_ERROR_PROVIDER_LIST_EXHAUSTED if the index is past the end of the list of endpointLists.
+     */
+    virtual CHIP_ERROR ReadEndpointListAtIndex(uint16_t index, EndpointListStorage & epList) = 0;
+
+    /**
+     * Check whether there is an action with the given actionId in the list of actions.
+     * @param actionId The action ID to search for.
+     * @return Returns a true if matching action is found otherwise false.
+     */
+    virtual bool HaveActionWithId(uint16_t actionId) = 0;
+
+    /**
+     * On receipt of each and every command,
+     * if the InvokeID data field is provided by the client when invoking a command, the server SHALL generate a StateChanged event
+     * when the action changes to a new state or an ActionFailed event when execution of the action fails.
+     *
+     * If the command refers to an action which currently is not in a state where the command applies, a response SHALL be
+     * generated with the Status::InvalidCommand.
+     */
+
+    /**
+     * When an InstantAction command is received, an action (state change) on the involved endpoints shall trigger,
+     * in a "fire and forget" manner. Afterwards, the action’s state SHALL be Inactive.
```

Comment:
The actual documentation should be something like "Handle an InstantAction command that was received", no?    Again, not sure copy/pasting the spec bits here is useful.... Similar for all the other commands here.